### PR TITLE
New test_quantize_conformance for HF

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Add conformance tests for bert-base-uncased  fro HF ( without validation)
- rework conformance tests:
   - `result.csv` rows splited by backends.
   - `results.csv` contains `metric name`, `metric value`, `quantization time` and `memory used in quantitation`.
   - each quantization pipeline by model/backends runs in separately test_case to have possible parallel on ci, and get more detailed pytest report.
   - Fail test by check with reference metric (atol=0.001, for some model metrics can be little difference)
   - Shows the improvement of the metrics in the status column and highlights green in result.html

### Related tickets
110630, 111488

### CODE REVIEW

File: tests/post_training/pipelines/masked_language_modeling.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    def get_transform_calibration_fn(self):
+        if self.backend in PT_BACKENDS:
+
+            def transform_func(data):
+                return torch.Tensor([data["input_ids"]]).type(dtype=torch.LongTensor)
+
+        else:
+
+            def transform_func(data):
+                print(data)
```
Comment (l-bat) [inline tests/post_training/pipelines/masked_language_modeling.py:56]: Do we really need this print?

File: tests/post_training/conftest.py [hunk 2]
```diff
@@ -9,219 +9,29 @@
...
-        "--eval_fp32",
-        action="store_true",
-        help="Evaluation fp32 model, by defaults used cached metric.",
-    )
-    parser.addoption(
-        "--do_bench",
-        action="store_true",
-        help="Collect performance statistics.",
-    )
+    # parser.addoption("--mode", action="store", default="short", choices=("full", "short", "no_val", "quantize_only"))
```
Comment (l-bat) [inline tests/post_training/conftest.py:27]: Remove?

File: tests/post_training/pipelines/masked_language_modeling.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+
+class MaskedLanguageModelingHF(BaseTestPipeline):
+    """Pipeline for masked language models from Hugging Face repository"""
+
+    def prepare_model(self) -> None:
+        if self.backend in PT_BACKENDS:
+            self.model_hf = transformers.AutoModelForSequenceClassification.from_pretrained(self.model_id)
+            self.model = self.model_hf
+            self.dummy_tensor = torch.Tensor([[0] * 53]).type(dtype=torch.LongTensor)
```
Comment (l-bat) [inline tests/post_training/pipelines/masked_language_modeling.py:34]: Can we store the magic value 53 in a constant/variable?

File: tests/post_training/test_quantize_conformance.py [hunk 4]
```diff
@@ -9,337 +9,21 @@
...
-    # Dump model
-    ov_path = Path(output_path) / (model_name + ".xml")
-    ov.serialize(model, str(ov_path))
-
-    # Benchmark performance
-    performance = benchmark_performance(ov_path, model_name, do_bench)
-    # Validate accuracy
-    accuracy = validate_accuracy(ov_path, dataloader)
-    return RunInfo(top_1=accuracy, fps=performance)
+# @pytest.fixture(scope="session", name="mode")
```
Comment (kshpv) [inline tests/post_training/test_quantize_conformance.py:24]: Please, remove commented code

**Review by kshpv — APPROVED** [review]
Please rem

File: tests/post_training/pipelines/masked_language_modeling.py (hunk 3 above)
Comment (AlexanderDokuchaev) [reply tests/post_training/pipelines/masked_language_modeling.py]: replaced to use dummy_inputs from HF model.

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

**Review by l-bat — APPROVED** [review]

**Pull request merged**